### PR TITLE
RTP tracks compositing example

### PR DESCRIPTION
**Pull request by chadbailey59**
This is based on some work I did for a customer. It has minimal customization to the UI, but it has a lot of JavaScript that shows how to submit a compositing job and wait on the result.

For the customer demo, I went back and forth on whether to do this in the browser or in a server process. For this example, I created a basic Daily API proxy and left it all in the browser. I could be convinced to move it to a Vercel serverless function, though.

I still need to do some cleanup, but I'm opening the PR in case anyone has better advice on how to organize all of this.

### CODE REVIEW

File: prebuilt-ui/rtp-tracks-compositing/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+- [Next.js Documentation](https://nextjs.org/docs) - learn about Next.js features and API.
+- [Learn Next.js](https://nextjs.org/learn) - an interactive Next.js tutorial.
+
+You can check out [the Next.js GitHub repository](https://github.com/vercel/next.js/) - your feedback and contributions are welcome!
+
+## Deploy on Vercel
+
+The easiest way to deploy your Next.js app is to use the [Vercel Platform](https://vercel.com/new?utm_medium=default-template&filter=next.js&utm_source=create-next-app&utm_campaign=create-next-app-readme) from the creators of Next.js.
+
+Check out our [Next.js deployment documentation](https://nextjs.org/docs/deployment) for more details.
```
Comment (philmillman) [inline prebuilt-ui/rtp-tracks-compositing/README.md:34]: this will need a real README

File: prebuilt-ui/rtp-tracks-compositing/pages/index.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        >
+          Powered by{' '}
+          <span className={styles.logo}>
+            <Image src="/vercel.svg" alt="Vercel Logo" width={72} height={16} />
+          </span>
+        </a>
+      </footer>
+    </div>
+  )
+}
```
Comment (philmillman) [inline prebuilt-ui/rtp-tracks-compositing/pages/index.js:69]: do we need the boilerplate index page?

File: prebuilt-ui/rtp-tracks-compositing/public/index.html [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+      		console.log('Recording stopped');
+      	});
+
+      	call.on('left-meeting', (e) => {
+      		// might want to just start processing the recording after the meeting starts in case the instructor closes the window
+      		// or use webhook-meeting-end to process recording
+      		console.log('Leaving meeting');
+      	});
+
+      	window.token = await createMeetingToken({
```
Comment (philmillman) [inline prebuilt-ui/rtp-tracks-compositing/public/index.html:52]: why the global?

File: prebuilt-ui/rtp-tracks-compositing/public/index.html [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+      	});
+
+      	let participants = await call.join({
+      		token: token
+      	});
+
+      	call.setLocalVideo(true);
+      	call.setActiveSpeakerMode(false);
+
+      	window.instructorSessionId = participants.local.session_id;
```
Comment (philmillman) [inline prebuilt-ui/rtp-tracks-compositing/public/index.html:66]: why the global?

File: prebuilt-ui/rtp-tracks-compositing/public/index.html [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+      		let rec = await getRecordingInfo(recordingId);
+      		// console.log("Recording info: ", rec);
+      		if (
+      			rec.tracks.find(e => e.type == "audio" && e.user_id == "instructor") &&
+      			rec.tracks.find(e => e.type == "video" && e.user_id == "instructor")
+      		) {
+      			return rec.tracks;
+      		} else {
+      			console.log("Waiting for tracks, currently: ", rec.tracks)
+      			await wait();
```
Comment (philmillman) [inline prebuilt-ui/rtp-tracks-compositing/public/index.html:128]: is this some sort of hack to do long polling of that endpoint?

File: prebuilt-ui/rtp-tracks-compositing/public/index.html [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+      	call.startRecording();
+      }
+
+      async function stopSession() {
+      	call.sendAppMessage({
+      		message: 'stopSession'
+      	}, '*');
+      	call.stopRecording();
+      	console.log('Starting processing for recording id:', recordingId);
+      	processRecording(￼)
```
Comment (philmillman) [inline prebuilt-ui/rtp-tracks-compositing/public/index.html:92]: image
I'm assuming this is an emoji?

**Review by philmillman — COMMENTED** [review]
I'm not sure I understand the advantage of using Next here, aside from the API proxy. This feels like something that could be done entirely on the server, unless the intent is to also demonstrate how to initiate the recording. 

From a general standards perspective there's a bunch of boilerplate that needs to be removed and everything needs to be linted/prettified.

File: prebuilt-ui/rtp-tracks-compositing/public/index.html (hunk 5 above)
Comment (chadbailey59) [reply prebuilt-ui/rtp-tracks-compositing/public/index.html:128]: Yes... once the call is over, we have to wait for the necessary tracks to appear in the track list, so we have to poll

**Comment by chadbailey59** [comment]
After careful consideration and thorough discussion, I have decided to move this to a secret location in Notion. :)

**Pull request closed without merging**